src/main/java/cope/cosmos/client/features/modules/movement/Step.java

```diff
@@ -42,57 +35,55 @@ public void onDisable() {
 
     @Override
     public void onUpdate() {
-        // reset vanilla step height
-        mc.player.stepHeight = 0.5F;
-
-        switch (mode.getValue()) {
-            case NORMAL: {
-                if (mc.player.onGround) {
-                    double stepHeight = 0.0; // the amount of blocks we have to step up
-
-                    for (double[] collisions : COLLISIONS) {
-                        if (isBoundingBoxEmpty(mc.player.motionX, collisions[0], mc.player.motionZ) && !isBoundingBoxEmpty(mc.player.motionX, collisions[1], mc.player.motionZ)) {
-                            stepHeight = Math.round(collisions[0] * 2.0) / 2.0; // round to nearest half
-                        }
-                    }
-
-                    // TODO: maybe use NCP lag compensation against itself? or find a way to do 1.5-2.5 step
-                    if (stepHeight != 0 && height.getValue() >= stepHeight) { // if we have blocks to step up and we're within our step height range
-                        double[] packetHeights = stepHeights.getOrDefault(stepHeight, null); // get our step heights from the map assigned to the block height
-
-                        if (packetHeights != null && packetHeights.length > 0) {
-                            for (double height : packetHeights) {
-                                mc.player.connection.sendPacket(new CPacketPlayer.Position(mc.player.posX, mc.player.posY + height, mc.player.posZ, mc.player.onGround));
-                            }
-
-                            // since we're ony sending packets, we're desynced with the server in our position. let's fix that by setting our clientside position
-                            mc.player.setPosition(mc.player.posX, mc.player.posY + stepHeight, mc.player.posZ);
-                        }
-                    }
-                }
-
-                break;
-            }
-
-            case VANILLA: {
-                // simple, vanilla step height.
-                mc.player.stepHeight = height.getValue().floatValue();
-                break;
-            }
-
-            case SPIDER: {
-                // honestly the anticheat has to be complete dog for this to work, but just in case
-                mc.player.motionY = 0.239;
-                break;
-            }
+        mc.player.stepHeight = height.getValue().floatValue();
+
+        if (timering)
+            getCosmos().getTickManager().setClientTicksDirect(50);
+    }
+
+    @SubscribeEvent
+    public void onStepEvent(StepEvent event) {
+
+        if (mode.getValue().equals(Mode.VANILLA))
+            return;
+
+        if (!mc.player.onGround) {
+            event.setHeight(.6f);
+            return;
         }
+
+        double step = event.getBB().minY - mc.player.posY;
+
+        if (step == 1)
```

Comment:
oh lord.

Reply:
if it works, it works

Reply:
bad mindset that wont get you anywhere 👎